src/repart/repart.c

```diff
@@ -2675,6 +2675,20 @@ static int determine_current_padding(
         return 0;
 }
 
+static int verify_regular_or_block(int fd) {
+        struct stat st;
+
+        assert(fd >= 0);
+
+        if (fstat(fd, &st) < 0)
+                return -errno;
+
+        if (!S_ISREG(st.st_mode) && !S_ISBLK(st.st_mode))
+                return -EBADFD;
```

Comment:
my suggestion was to return EISDIR in case of a dir, and ELOOP in case of a symlink, i.e. what stat_verify_regular() does

Reply:
Sure, can do. But is `-EBADFD` an appropriate error to propagate in this case? Or is something else preferable?

Reply:
Latest push handles the dir/link cases, please let me know if the last error should change as well.